src/style/colors.js

```diff
@@ -81,22 +77,52 @@ class Colors {
         return color;
       }
       return requiredColor;
-    } else { // Handles dynamic colors (non uilib colors)
-      let tintLevel = Math.floor(Number(tintKey) / 10);
-      tintLevel = Math.max(1, tintLevel);
-      tintLevel = Math.min(8, tintLevel);
-      
-      if (tintLevel === BASE_COLOR_LEVEL) {
-        return color;
-      } else if (tintLevel <= BASE_COLOR_LEVEL) {
-        const darkRatio = darkRatios[tintLevel - 1];
-        return Color(color).darken(darkRatio).hex();
-      } else {
-        const lightRatio = lightRatios[tintLevel - 4];
-        return Color(color).mix(Color('#ffffff'), lightRatio).hex();
-      }
+    } else { 
+      // Handles dynamic colors (non uilib colors)
+      let tintLevel = Math.floor((Number(tintKey) / 10) + 1);
+      tintLevel = Math.max(2, tintLevel);
+      tintLevel = Math.min(9, tintLevel);
+      return generateColorTint(color, tintLevel * 10);
     }
   }
+
+  generateColorPalette(color) {
+    const hsl = Color(color).hsl();
+    const lightness = Math.round(hsl.color[2]);
+    
+    const ls = [lightness];
+    let l = lightness;
+    do {
+      l -= 10;
+      if (l > 20) {
```

Comment:
what's the point of having the condition `l > 20` twice? in the `while` and in the `if` statements
instead, you can initialize `l` with `lightness - 10`  and do a simple while

Reply:
but it will still do the last loop iteration, which I don't want in my array...

Reply:
are you sure?  if you do the following
```
let l = lightness -10;
while (l > 20) {
  ls.push(l);
  l -= 10;
}
```
This doesn't do the same?

Reply:
Done.